src/System.Runtime/tests/System/Text/RuneTests.netcoreapp.cs

```diff
@@ -542,6 +541,12 @@ public static void TryEncode_AsUtf16(GeneralTestData testData)
             Assert.False(success);
             Assert.Equal(0, charsWritten);
 
+            Assert.Throws<ArgumentException>(() =>
+            {
+                Span<char> utf16Buffer = stackalloc char[rune.Utf16SequenceLength - 1];
+                rune.Encode(utf16Buffer);
```

Comment:
Nit: it'd be a bit simpler as just:
```C#
Assert.Throws<ArgumentException>(() => rune.Encode(new char[rune.Utf16SequenceLength - 1]));
```
Also, is there an argument name we can validate with AssertExtensions.Throws?

Reply:
There's no argument name in the exception message right now. (It's the same behavior as `Span<T>.CopyTo`.) I can certainly plumb an argument name through the exception message easily enough.